### PR TITLE
fix: increase trial log timestamp resolution to support milliseconds [DET-4861]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

When there are a lot of trial logs happening with a short time of 1 second, the log viewer can potentially stop rendering older logs. The issue is that the requests are based on timestamps and with only resolution of whole seconds, the request for older trial logs remain the same, making it seem like the trial logs has terminated at that timestamp.

Update the trial log requests to use timestamps with milliseconds resolution.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Review by hamidzr — COMMENTED** [review]
Thanks for working on this. This issue of not being able to index logs with timestamps accurately is this a known limitation that we have to deal with when using Elastic? 
 Could we still have the issue even with milliseconds?

build fails for me, I tried a fresh run of the codegen but it didn't help. should be something simple. (fails in ci as well)
```
Type '{ id: string; level: string; message: string; time: string; }' is not assignable to type 'TrialLog'.
  Types of property 'level' are incompatible.

```

File: proto/src/determined/api/v1/trial.proto [hunk 1]
```diff
@@ -49,7 +49,7 @@ message TrialLogsRequest {
 // Response to TrialLogsRequest.
 message TrialLogsResponse {
   option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
-    json_schema: { required: [ "id", "message" ] }
+    json_schema: { required: [ "id", "level", "message", "timestamp" ] }
```
[the remark this answers is not in the document]
Comment (hkang1) [reply proto/src/determined/api/v1/trial.proto:52]: @stoksc confirmed that with both backings, both `level` and `timestamp` will exist

File: webui/react/src/components/LogViewerTimestamp.tsx [hunk 2]
```diff
@@ -27,8 +27,8 @@ import Page, { Props as PageProps } from './Page';
 
 export interface LogViewerTimestampFilter {
   limit?: number,
-  timestampAfter?: Dayjs,
-  timestampBefore?: Dayjs,
+  timestampAfter?: Dayjs,   // exclusive of the specified date time
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/components/LogViewerTimestamp.tsx:30]: I've seen it both ways, but I think you're right, the way you described is more common. This is just a mapping to the api parameters however.

**Comment by hkang1** [comment]
> Thanks for working on this. This issue of not being able to index logs with timestamps accurately is this a known limitation that we have to deal with when using Elastic?
>
I'm not sure I follow. The only difference between the current implementation between the elastic-backed and postgresql is the id where elastic will be a hexadecimal string and postrgresql is a number as a string. But the ordering of the logs is based on the order that comes from the endpoint, which the endpoint bases on the timestamp + the order_by query param (asc vs desc).

> Could we still have the issue even with milliseconds?
> 
I forgot to make the limit larger (from 100) but with a higher limit (thinking 500) and milliseconds I think that makes it pretty unlikely it will happen. It would require 500 events to occur within the same millisecond

**Comment by hamidzr** [comment]
> where elastic will be a hexadecimal string and postrgresql is a number as a string

I'm talking about this part. Would it be accurate to say with Postgres we are getting definite ordering with a unique monotonically increasing integer whereas since that order doesn't exist in the same level in elastic we fall back to using timestamps which can have duplicates

**Review by hamidzr — APPROVED** [review]
LGTM

File: webui/react/src/services/apiConfig.ts [hunk 3]
```diff
@@ -415,5 +415,5 @@ export const getTrialLogs: HttpApi<TrialLogsParams, TrialLog[]> = {
     ].join('?'),
   }),
   name: 'getTrialLogs',
-  postProcess: response => jsonToTrialLogs(response.data),
+  postProcess: response => jsonToTrialLogs(response.data as Api.V1TrialLogsResponse[]),
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/react/src/services/apiConfig.ts]: this seems outdated now

**Comment by hkang1** [comment]
@hamidzr thanks for the quick review as usual!

**Pull request merged**